### PR TITLE
add new stm32 nucleo l552ze board

### PR DESCRIPTION
**Pull request by FRASTM**
add the new **NUCLEO-L552ZE-Q**
STM32 Nucleo-144 development board with STM32L552ZE MCU, SMPS, supports Arduino, ST Zio and morpho connectivity  144 hardware supporting the stm32L5552ZE device.

Note that USER_BTN has external PullDown and `pinMode(USER_BTN, INPUT);` is preferred 

See [NUCLEO-L552ZE-Q](https://www.st.com/content/st_com/en/products/evaluation-tools/product-evaluation-tools/mcu-mpu-eval-tools/stm32-mcu-mpu-eval-tools/stm32-nucleo-boards/nucleo-l552ze-q.html)
Requires the https://github.com/stm32duino/Arduino_Core_STM32/pull/1249

Signed-off-by: Francois Ramu francois.ramu@st.com

### CODE REVIEW

**Comment by FRASTM** [comment]
Fix I2C pins : SDA on PB9 and SCL on PB8
--> X_NUCLEO_IKS01A3_STTS751_TemperatureLimit.ino  OK

**Comment by FRASTM** [comment]
tested: SPI_A loopback transfer OK  (A_MISO connected to A_MOSI) with a sample test application

**Comment by FRASTM** [comment]
rebase
Test EEPROM put/get : passed
Test HWTimer  : passed
Test SerialLoop : passed (with USART3 on pin PD9, PD8)

**Comment by FRASTM** [comment]
add the support of USB FS : serial CD and Mouse&HID : passed

File: cores/arduino/stm32/timer.h [hunk 1]
```diff
@@ -63,17 +66,19 @@ extern "C" {
...
 #if defined(TIM6_BASE) && !defined(TIM6_IRQn)
 #if defined(DAC_BASE) || defined(DAC1_BASE)
 #if defined(STM32G0xx)
 #define TIM6_IRQn TIM6_DAC_LPTIM1_IRQn
 #define TIM6_IRQHandler TIM6_DAC_LPTIM1_IRQHandler
-#elif !defined(STM32F1xx) && !defined(STM32L1xx) && !defined(STM32MP1xx)
+#elif !defined(STM32F1xx) && !defined(STM32L1xx) && !defined(STM32L5xx) && !defined(STM32MP1xx)
 #define TIM6_IRQn TIM6_DAC_IRQn
-#define TIM6_IRQHandler TIM6_DAC_IRQHandler
+#define TIM6_IRQHandler TIM2_DAC_IRQHandler
```
Comment (fpistm) [inline cores/arduino/stm32/timer.h:77]: Why change to `TIM2_DAC_IRQHandler`

**Review by fpistm — CHANGES_REQUESTED** [review]

File: cores/arduino/stm32/timer.h (hunk 1 above)
Comment (fpistm) [reply cores/arduino/stm32/timer.h:77]: Moreover TIM2_DAC_IRQHandler does not exist.

File: variants/NUCLEO_L552ZE_Q/variant.cpp [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+  RCC_ClkInitStruct.SYSCLKSource = RCC_SYSCLKSOURCE_PLLCLK;
+  RCC_ClkInitStruct.AHBCLKDivider = RCC_SYSCLK_DIV1;
+  RCC_ClkInitStruct.APB1CLKDivider = RCC_HCLK_DIV1;
+  RCC_ClkInitStruct.APB2CLKDivider = RCC_HCLK_DIV1;
+
+  if (HAL_RCC_ClockConfig(&RCC_ClkInitStruct, FLASH_LATENCY_5) != HAL_OK) {
+    Error_Handler();
+  }
+}
+
```
Comment (fpistm) [inline variants/NUCLEO_L552ZE_Q/variant.cpp:218]: USB clock config is missing

File: variants/NUCLEO_L552ZE_Q/variant.h [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+#define PB0                     A3
+#define PC1                     A4
+#define PC0                     A5
+#define PB1                     A6
+#define PC2                     A7
+#define PA1                     A8
+
+#define PA0_ALT0                A14
+#define PA1_ALT0                A15
+#define PA1_ALT1                113
```
Comment (fpistm) [inline variants/NUCLEO_L552ZE_Q/variant.h:144]: should be 118, pin numbering is wrong.

File: variants/NUCLEO_L552ZE_Q/variant.h [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+#define PC3_ALT0                A27
+#define PC6_ALT0                147
+#define PC7_ALT0                148
+#define PC8_ALT0                149
+#define PC9_ALT0                150
+#define PC10_ALT0               151
+#define PC11_ALT0               152
+#define PF9_ALT0                153
+
+#define NUM_DIGITAL_PINS        154
```
Comment (fpistm) [inline variants/NUCLEO_L552ZE_Q/variant.h:186]: Wrong value

File: variants/NUCLEO_L552ZE_Q/variant.h [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+#define PIN_SERIAL_RX           PG8
+#endif
+#ifndef PIN_SERIAL_TX
+#define PIN_SERIAL_TX           PG7
+#endif
+
+// Extra HAL modules
+#define HAL_DAC_MODULE_ENABLED
+#define HAL_SD_MODULE_ENABLED
+
```
Comment (fpistm) [inline variants/NUCLEO_L552ZE_Q/variant.h:260]: OSPI missing

File: variants/NUCLEO_L552ZE_Q/variant.h [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+#define PC6  16
+#define PD11 17
+#define PB13 18
+#define PD12 19
+#define PA4 20
+#define PB4 21
+#define PB5 22 // SPI_B_MOSI
+#define PB3 23 // SPI_B_SCK
+#define PA4 24 // SPI_B_NSS
+#define PB4 25 // SPI_B_MISO
```
Comment (fpistm) [inline variants/NUCLEO_L552ZE_Q/variant.h:48]: Multiple definition, several warning about that during the compile
// 24 is PA4                   (20)
// 25 is PB4                   (21)

Other pins are also duplcated

File: boards.txt [hunk 7]
```diff
@@ -163,6 +163,19 @@ Nucleo_144.menu.pnum.NUCLEO_L4R5ZI-P.build.product_line=STM32L4R5xx
...
+Nucleo_144.menu.pnum.NUCLEO_L552ZE_Q.node=NODE_L552ZE
+Nucleo_144.menu.pnum.NUCLEO_L552ZE_Q.upload.maximum_size=524288
+Nucleo_144.menu.pnum.NUCLEO_L552ZE_Q.upload.maximum_data_size=196608
+Nucleo_144.menu.pnum.NUCLEO_L552ZE_Q.build.mcu=cortex-m33
+Nucleo_144.menu.pnum.NUCLEO_L552ZE_Q.build.flags.fp=-mfpu=fpv4-sp-d16 -mfloat-abi=hard
+Nucleo_144.menu.pnum.NUCLEO_L552ZE_Q.build.board=NUCLEO_L552ZE_Q
+Nucleo_144.menu.pnum.NUCLEO_L552ZE_Q.build.series=STM32L5xx
+Nucleo_144.menu.pnum.NUCLEO_L552ZE_Q.build.product_line=STM32L552xx
+Nucleo_144.menu.pnum.NUCLEO_L552ZE_Q.build.variant=NUCLEO_L552ZE_Q
+Nucleo_144.menu.pnum.NUCLEO_L552ZE_Q.build.cmsis_lib_gcc=arm_cortexM33lf_math
```
Comment (fpistm) [inline boards.txt:177]: Where are this library ?

**Comment by fpistm** [comment]
Superseded by #1091

**Pull request closed without merging**